.changeset/swift-spoons-act.md

```diff
@@ -0,0 +1,103 @@
+---
+"thirdweb": minor
+---
+
+Get NFTs by a wallet address.
+```ts
+import { createThirdwebClient, getNFTsByOwnert } from "thirdweb";
+
+const client = createThirdwebClient({ clientId: "..." });
+const nfts = await getNFTsByOwner({
+ client,
+ owner: "0x...",
+ pageSize: 20,
+ page: 1
+});
+```
+
+---
+
+Get NFTs by collection.
+```ts
+import { createThirdwebClient, defineChain, getNFTsByCollection, getContract } from "thirdweb";
+
+const client = createThirdwebClient({ clientId: "..." });
+const contract = getContract({
+ address: "0x...",
+ chain: defineChain(1)
+});
+const nfts = await getNFTsByCollection({
+ contract,
+ groupBy: ["ownerAddress"],
+});
+```
+
+---
+
+Get latest block number for a chain.
+```ts
+import { createThirdwebClient, getLatestBlockNumber } from "thirdweb";
+
+const client = createThirdwebClient({ clientId: "..." });
+const { latestBlockNumber } = await getLatestBlockNumber({
+ client,
+ chain: defineChain(1)
+});
+```
+
+---
+
+Get events for a contract.
+```ts
+import { createThirdwebClient, getEvents, defineChain, getContract } from "thirdweb";
+
+const startDate = new Date(Date.now() - 48*60*60_000);
+const endDate = new Date();
+const contract = getContract({
+ address: "0x...",
+ client,
+ chain: defineChain(1)
+});
+const events = await getEvents({
```

Comment:
should we build this into the existing getContractEvents that we have?

Reply:
Yeah we should. Right now the indexer gets events based on start and end time though and `getContractEvents` does it based on block numbers. Let me ask if we can add block number filter options on the indexer, but we won't be able to fallback to RPC call when filtering by times